dali/operators/decoder/audio/generic_decoder.cc

```diff
@@ -155,104 +135,99 @@ AudioMetadata GetAudioMetadata(const SF_INFO &sf_info) {
 }  // namespace
 
 
-template<typename SampleType>
-ptrdiff_t GenericAudioDecoder<SampleType>::DecodeTyped(span<SampleType> output) {
-  return impl_->DecodeTyped(output);
-}
-
-
-template<typename SampleType>
-AudioMetadata GenericAudioDecoder<SampleType>::OpenImpl(span<const char> encoded) {
-  return impl_->OpenImpl(encoded);
-}
-
-template<typename SampleType>
-AudioMetadata GenericAudioDecoder<SampleType>::OpenFromFileImpl(const std::string &filepath) {
-  return impl_->OpenFromFileImpl(filepath);
-}
-
-template<typename SampleType>
-void GenericAudioDecoder<SampleType>::CloseImpl() {
-  impl_->CloseImpl();
-}
-
-
-template<typename SampleType>
-GenericAudioDecoder<SampleType>::~GenericAudioDecoder() = default;
+class GenericAudioDecoder : public AudioDecoderBase {
+ public:
+  GenericAudioDecoder() = default;
+  GenericAudioDecoder(GenericAudioDecoder&&) = default;
+  GenericAudioDecoder& operator=(GenericAudioDecoder&&) = default;
```

Comment:
Is this correct? I believe that at least `sndfile_handle_` should be reset. I don't know how remaining fields should be moved